apps/src/labs/LevelChangeManager.ts

```diff
@@ -18,24 +18,30 @@ export default class LevelChangeManager {
    * Change levels and handle level reset.
    * @param existingProject The current project to save before changing levels (if it has changed).
    * @param existingProjectManager The current project manager to clean up before changing levels.
+   * @param defaultCode Default code for the new level.
    * @param levelId The level id of the new level.
    * @param scriptId The script id of the new level (optional).
    */
   async changeLevel(
     existingProject: Project,
     existingProjectManager: ProjectManager,
+    defaultCode: object,
     levelId: number,
     scriptId?: number
   ) {
     if (existingProjectManager.hasUnsavedChanges()) {
       // Force a save with the current code before changing levels if there are unsaved changes.
       await existingProjectManager.save(existingProject, true);
     }
-    // Clear out any remaining enqueud saves from the existing project manager.
+    // Clear out any remaining enqueued saves from the existing project manager.
     existingProjectManager.destroy();
     // Reset the level.
-    await this.handleLevelReset(levelId, scriptId);
+    await this.handleLevelReset(defaultCode, levelId, scriptId);
   }
 }
 
-type LevelResetHandler = (levelId: number, scriptId?: number) => Promise<void>;
+type LevelResetHandler = (
+  defaultCode: object,
```

Comment:
We definitely want to be given some start sources in case the server returns a `404`.  I'm not sure whether to enforce here but it would help us catch missing cases if we leave it as required.

Reply:
There are cases where a level has no start sources, right? (per the level definition anyway) In another function you commented that start sources were optional

Reply:
I removed the comment that they were optional. :)  A level should have start sources, or we won't even have a `when_run` block, so for now we want them to always be provided.